apim-operator/pkg/controller/api/api_controller.go

```diff
@@ -259,3 +259,88 @@ func createMGWSecret(r *ReconcileAPI, confData string) error {
 	}
 
 }
+
+//microgateway deployment within init container
+func createMicroGatewayDeployment(cr *wso2v1alpha1.API) *corev1.Pod {
+	labels := map[string]string{
+		"app": cr.Name,
+	}
+
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-pod",
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: corev1.PodSpec{
+			InitContainers: []corev1.Container{
+				{
+					Name:  "gen-balx",
+					Image: "dinushad/bal:v3",
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "swagger-volume",
+							MountPath: "/usr/wso2/swagger/",
+							ReadOnly:  true,
+						},
+						{
+							Name:      "mgw-volume",
+							MountPath: "/usr/wso2/mgw/",
+						},
+						{
+							Name:      "balx-volume",
+							MountPath: "/home/exec/",
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:  "micro-gateway",
+					Image: "wso2/wso2micro-gw:3.0.0-beta2",
```

Comment:
We need to move those configurations to a configmap. Then we can read those configmap and populate those values.

Reply:
Ack. I will add it as a todo